Chatkit/Redux/Store/StoreBroadcaster.swift

```diff
@@ -0,0 +1,48 @@
+
+protocol StoreListener: AnyObject { // AnyObject is neccessary to use `===` operator
+    func store(_ store: Store, didUpdateState state: State)
+}
+
+protocol HasStoreBroadcaster {
+    var storeBroadcaster: StoreBroadcaster { get }
+}
+
+protocol StoreBroadcaster: StoreDelegate {
+    func register(_ listener: StoreListener) -> State
+    func unregister(_ listener: StoreListener)
+}
+
+class ConcreteStoreBroadcaster: StoreBroadcaster {
+    
+    typealias Dependencies = HasStore
+    
+    private let dependencies: Dependencies
+    
+    private var listeners = [StoreListener]()
+    
+    init(dependencies: Dependencies) {
+        self.dependencies = dependencies
+    }
+    
+    // MARK: StoreBroadcaster
+    
+    func register(_ listener: StoreListener) -> State {
```

Comment:
Why do we return `State` from this method?

Reply:
Otherwise the object thats registering has no way of knowing the state until the next time the state changes and the delegate method is invoked.